website/server/libs/inbox/index.js

```diff
@@ -83,6 +83,18 @@ export async function getUserInboxMessage (user, messageId) {
   return Inbox.findOne({ ownerId: user._id, _id: messageId }).exec();
 }
 
+export async function getInboxMessageOfOppositeTarget (user, messageId) {
+  const usersMessage = await getUserInboxMessage(user, messageId);
+
+  return Inbox.findOne({
+    ownerId: usersMessage.uuid,
+    uuid: user._id,
+    text: usersMessage.text,
+    // timestamp not possible (ms difference..)
+    // id not possible (different ID per same message across both owners)
```

Comment:
whuf, this seems like a really dubious way to find the message
won't this lead to inconsistent results if a user sends an identical message in the same conversation?
we could save a new UUID field that's the same on both copies of the message, instead.

Reply:
yeah... we definitely need a unique ID per message, with the current system it just creates headaches 😅 

I'll try to create the migration script for that which we'd have to run as well